TypeScript compilation errors after updating `@uppy/onedrive` to 4.1.2

### Initial checklist

- [X] I understand this is a bug report and questions should be posted in the [Community Forum](https://community.transloadit.com/)
- [X] I searched [issues](https://github.com/transloadit/uppy/issues?q=is%3Aissue) and couldn’t find anything (or linked relevant results below)

### Link to runnable example

_No response_

### Steps to reproduce

Our Vite/React project started failing to build after updating Uppy packages to their latest versions:

```
node_modules/@uppy/core/src/Uppy.ts:35:36 - error TS5097: An import path can only end with a '.ts' extension when 'allowImportingTsExtensions' is enabled.

35 import supportsUploadProgress from './supportsUploadProgress.ts'
                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@uppy/core/src/Uppy.ts:36:25 - error TS5097: An import path can only end with a '.ts' extension when 'allowImportingTsExtensions' is enabled.

36 import getFileName from './getFileName.ts'
                           ~~~~~~~~~~~~~~~~~~

node_modules/@uppy/core/src/Uppy.ts:37:47 - error TS5097: An import path can only end with a '.ts' extension when 'allowImportingTsExtensions' is enabled.

37 import { justErrorsLogger, debugLogger } from './loggers.ts'
                                                 ~~~~~~~~~~~~~~

node_modules/@uppy/core/src/Uppy.ts:42:8 - error TS5097: An import path can only end with a '.ts' extension when 'allowImportingTsExtensions' is enabled.

42 } from './Restricter.ts'
          ~~~~~~~~~~~~~~~~~

node_modules/@uppy/core/src/Uppy.ts:46:20 - error TS5097: An import path can only end with a '.ts' extension when 'allowImportingTsExtensions' is enabled.

46 import locale from './locale.ts'
                      ~~~~~~~~~~~~~

node_modules/@uppy/core/src/Uppy.ts:652:9 - error TS2578: Unused '@ts-expect-error' directive.

652         // @ts-expect-error these typed are inserted
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@uppy/core/src/Uppy.ts:902:14 - error TS18046: 'err' is of type 'unknown'.

902       return err.message
                 ~~~

node_modules/@uppy/core/src/Uppy.ts:914:14 - error TS18046: 'err' is of type 'unknown'.

914       return err.message
                 ~~~


Found 8 errors in the same file, starting at: node_modules/@uppy/core/src/Uppy.ts:35
```

It seemed odd that building our project would try to build TS files in `node_modules/@uppy/core/src`.

After downgrading each Uppy package individually until the build failure was resolved, we isolated the issue to `@uppy/onedrive` version 4.1.2. Being on the latest version of all `@uppy` packages but with `onedrive 4.1.1` works fine.

It _seems_ like this `import` might be the culprit:

https://github.com/transloadit/uppy/blob/main/packages/%40uppy/onedrive/src/OneDrive.tsx#L11

It imports from `src` and not `lib`.

This line was recently introduced in https://github.com/transloadit/uppy/pull/5443, which was released as `@uppy/onedrive` version 4.1.2 (the faulty version).

### Expected behavior

Project should build without any errors.

### Actual behavior

Project builds with errors.

--- a/packages/@uppy/onedrive/src/OneDrive.tsx
+++ b/packages/@uppy/onedrive/src/OneDrive.tsx
@@ -8,9 +8,9 @@ import { UIPlugin, Uppy } from '@uppy/core'
 import { ProviderViews } from '@uppy/provider-views'
 import { h, type ComponentChild } from 'preact'
 
-import type { AsyncStore } from '@uppy/core/src/Uppy.js'
 import type { UppyFile, Body, Meta } from '@uppy/utils/lib/UppyFile'
 import type {
+  AsyncStore,
   UnknownProviderPlugin,
   UnknownProviderPluginState,
 } from '@uppy/core/lib/Uppy.js'

